test/analysis_tests.py

```diff
@@ -1009,6 +1035,414 @@ def test_loss(self):
 		loss = loss_class.loss(yttf,yptf)
 
 		self.assertAlmostEqual(np.sum(loss.numpy()),scipy_nlp,places=4)
+		
+
+class FullCovarianceLossTests(unittest.TestCase):
```

Comment:
Not sure why this is marked as new, but I'm assuming the entire class is the same as before (and therefore doesn't need review).